client/asset/interface.go

```diff
@@ -418,6 +424,25 @@ type TxFeeEstimator interface {
 	EstimateSendTxFee(address string, value, feeRate uint64, subtract bool) (fee uint64, isValidAddress bool, err error)
 }
 
+// Bonder is a wallet capable of creating and redeeming time-locked fidelity
+// bond transaction outputs.
+type Bonder interface {
+	// MakeBondTx authors a DEX time-locked fidelity bond transaction for the
+	// provided amount, lock time, and dex account ID. An explicit private key
+	// type is used to guarantee it's not bytes from something else like a
+	// public key.
+	MakeBondTx(ver uint16, amt uint64, lockTime time.Time, privKey *secp256k1.PrivateKey, acctID []byte) (*Bond, error)
+	// RefundBond will refund the bond given the full bond output details and
+	// private key to spend it.
+	RefundBond(ctx context.Context, ver uint16, coinID, script []byte, amt uint64, privKey *secp256k1.PrivateKey) ([]byte, error)
+
+	// A RefundBondByCoinID may be created in the future to attempt to refund a
+	// bond by locating it on chain, i.e. without providing the amount or
+	// script, while also verifying the bond output is unspent. However, it's
+	// far more straightforward to generate the refund transaction using the
+	// known values.
+}
```

Comment:
I think we should also include methods for 1) locking and 2) renewing eventually.